### ISSUE TITLE
On register page required fields should be labeled as such prior to submit

### ISSUE THREAD

**Issue by htbox-tester** [HTBox/crisischeckin#26]
It would be better design if the required fields on the new user registration indicated which fields were required prior to submit.

**Comment by tditiecher** [comment]
To broaden the scope of this issue: this doesn't apply to only the user registration page but to all pages.

A simple and common solution is to put a red asterisk after the label for a required field. Is this what you want?

The simplest solution is to add a required class to the css and add it to the form-group div for all required fields:

```html
<style>
.required label:after {
    content: "*";
    padding-left: 5px;
    color: #C33A3A;
}
</style>

<div class="form-group required">
  @Html.LabelFor(m => m.FirstName)
  @Html.TextBoxFor(m => m.FirstName, new { @class = "form-control" })
</div>
```

The only disadvantage is that you have to mark both your viewmodel and your html field as required. A more complex solution would involve creating a custom html extension method that inspects the Required attribute on the viewmodel properties and add the required css class in code (something in the line of http://weblogs.asp.net/gunnarpeipman/asp-net-mvc-how-to-show-asterisk-after-required-field-label). But I think this way is a bit of overkill for the problem at hand. What do you think?

Another question: the file Forms.min.css is included in _Layout.cshtml but I don't see any bundles defined for minification of the css (and js). How is this file generated from Forms.css?

**Comment by BillWagner** [comment]
I'd prefer the simpler solutions. 

I'm not sure on your second question. I'd need to dive through the code. I can do that later, unless you've already found the answer.

**Comment by tditiecher** [comment]
Ok I'll add the required asterisk on all relevant fields in the forms where it makes sense (not only user registration).

About the Forms.css vs Forms.min.css, I didn't find the answer. I could just copy the css to both files and create a separate issue for this.

**Comment by BillWagner** [comment]
Fixed in pull request #250
